backend/src/estark/mod.rs

```diff
@@ -0,0 +1,223 @@
+
+    Ok(())
+}
+
+impl<'a, F: FieldElement> EStarkFileDumper<'a, F> {
+    fn create(
+        analyzed: &'a Analyzed<F>,
+        fixed: &'a [(String, Vec<F>)],
+        output_dir: Option<&'a Path>,
+        setup: Option<&mut dyn std::io::Read>,
+        verification_key: Option<&mut dyn std::io::Read>,
+        just_dump: bool,
+    ) -> Result<Box<dyn crate::Backend<'a, F> + 'a>, Error> {
+        if setup.is_some() {
+            return Err(Error::NoSetupAvailable);
+        }
+        if verification_key.is_some() {
+            return Err(Error::NoVerificationAvailable);
+        }
+
+        // Pre-process the PIL and fixed columns.
+        let (pil, fixed) = pil_hack_fix(analyzed, fixed);
+
+        Ok(Box::new(EStarkFileDumper {
+            degree: analyzed.degree(),
+            pil,
+            fixed,
+            output_dir,
+            just_dump,
+        }))
+    }
+}
+
+impl<'a, F: FieldElement> Backend<'a, F> for EStarkFileDumper<'a, F> {
+    fn prove(
+        &self,
+        witness: &[(String, Vec<F>)],
+        prev_proof: Option<Proof>,
+        // TODO: Implement challenges
+        _witgen_callback: WitgenCallback<F>,
+    ) -> Result<Proof, Error> {
+        if prev_proof.is_some() {
+            return Err(Error::NoAggregationAvailable);
+        }
+
+        // Write the files in the format expected by the zkvm-prover.
+        if let Some(output_dir) = self.output_dir {
+            // Write the constants.
+            let constants_path = output_dir.join("constants.bin");
+            log::info!("Writing {}.", constants_path.to_string_lossy());
+            write_polys_bin(&constants_path, &self.fixed)?;
+
+            // Write the commits.
+            let commits_path = output_dir.join("commits.bin");
+            log::info!("Writing {}.", commits_path.to_string_lossy());
+            write_polys_bin(&commits_path, witness)?;
+
+            // Write the stark struct JSON.
+            let stark_struct_path = output_dir.join("starkstruct.json");
+            log::info!("Writing {}.", stark_struct_path.to_string_lossy());
+            write_json_file(&stark_struct_path, &create_stark_struct(self.degree))?;
+
+            // Write the constraints in JSON.
+            let contraints_path = output_dir.join("constraints.json");
+            log::info!("Writing {}.", contraints_path.to_string_lossy());
+            write_json_file(&contraints_path, &self.pil)?;
+
+            if self.just_dump {
+                Ok(Vec::new())
+            } else {
+                // Generate the proof.
+                #[cfg(feature = "estark-polygon")]
+                return polygon_wrapper::prove_and_verify(
+                    &contraints_path,
+                    &stark_struct_path,
+                    &constants_path,
+                    &commits_path,
+                    output_dir,
+                );
+
+                #[cfg(not(feature = "estark-polygon"))]
+                Ok(Vec::new())
```

Comment:
Why doesn't it use the starky prover?

Reply:
This backend struct is used both by EStarkPolygon and EStarkDump. If feature "estark-polygon" is disabled, this can only be a EStarkDump, which doesn't do any proving, just dump the files.